src/renderer/qssWidget/js/qssOpenUsbWidget.js

```diff
@@ -0,0 +1,350 @@
+            return {
+                key: key,
+                value: values[index]
+            };
+        });
+    };
+
+    /**
+     * Gets the value the pressed button from the menu and determines which even to fire
+     * in case of no valid button type is pressed does nothing
+     *
+     * @param {String} value - accepts only "Mount" and "Unmount"
+     * @param {Component} openUSB - The `gpii.qssWidget.openUSB` instance
+     * @param {KeyboardEvent} keyboardEvent - The keyboard event (if any) that led to the
+     * change in the setting's value.
+     * @param {EventListener} mountUsbEvent - handle to the onQssOpenUsbRequested event
+     * @param {EventListener} unmountUsbEvent - handle to the onQssUnmountUsbRequested event
+     * @param {String} messageChannel - The channel to which the message should be sent.
+     */
+    gpii.qssWidget.openUSB.handleOpenUSB = function (value, openUSB, keyboardEvent, mountUsbEvent, unmountUsbEvent, messageChannel) {
+        if (fluid.isValue(value)) {
+            switch (value) {
+            case "Mount":
+                // fires the event that mounts and open the USB drive
+                mountUsbEvent.fire(messageChannel, openUSB.model.messages);
+                break;
+            case "Unmount":
+                // fires the event that ejects any attached USB drive
+                unmountUsbEvent.fire(messageChannel, openUSB.model.messages);
+                break;
+            default:
+                // do nothing in every other case
+                break;
+            }
+            openUSB.close(keyboardEvent);
+        }
+    };
+
+    /**
+     * A handler for the `repeater` instance in the QSS widget menu. Takes care of rendering
+     * a particular setting option and handling user interaction.
+     */
+    fluid.defaults("gpii.qssWidget.openUSB.presenter", {
+        gradeNames: ["fluid.viewComponent", "gpii.qssWidget.button"],
+        model: {
+            item: null
+        },
+        styles: {
+            active: "fl-qssWidgetMenu-active",
+            default: "fl-qssWidgetMenu-default"
+        },
+        modelListeners: {
+            item: {
+                this: "{that}.container",
+                method: "text",
+                args: ["{that}.model.item.value"]
+            },
+            "{repeater}.model.value": [{
+                funcName: "gpii.qssWidget.openUSB.presenter.animateActivation",
+                args: ["{change}.value", "{that}.model.item", "{that}.container", "{that}.options.styles"],
+                includeSource: "settingAlter"
+            }]
+        },
+        events: {
+            onItemFocus: "{repeater}.events.onItemFocus"
+        },
+        listeners: {
+            "onCreate.applyStyles": {
+                funcName: "gpii.qssWidget.openUSB.presenter.applyStyles",
+                args: ["{that}", "{that}.container", "{repeater}.model.styles"]
+            },
+            onItemFocus: {
```

Comment:
I notice that this listener is now repeated identically in 3 places - qssMenuWidget.js, qssScreenCaptureWidget.js and now here - time to factor this and all the things it depends on into a shared grade